neo4j/session_with_context.go

```diff
@@ -521,7 +538,9 @@ func (s *sessionWithContext) Run(ctx context.Context,
 	s.autocommitTx = &autocommitTransaction{
 		conn: conn,
 		res: newResultWithContext(conn, stream, cypher, params, func() {
-			s.retrieveBookmarks(conn, runBookmarks)
+			if err := s.retrieveBookmarks(ctx, conn, runBookmarks); err != nil {
+				s.log.Warnf(log.Session, s.logId, "could not retrieve bookmarks after result consumption: %s", err.Error())
```

Comment:
Maybe add explanation what this means for the user.  Along the lines of "this means that the query in question will not be part of the causal chain tracked by the provided bookmark manager" or what have you.

Reply:
fixed in ac05321b24e730f053d07a3e1dd6d69ca8b1b0ed